packages/lib-classifier/src/plugins/tasks/experimental/highlighter/components/HighlighterTask.js

```diff
@@ -0,0 +1,188 @@
+`
+
+const StyledButton = styled(Button)`
+  ${props => props.theme.dark ?
+    css`
+      background: transparent;
+      border: 2px solid ${props.theme.global.colors['light-1']};
+    ` :
+    css`
+      background: ${props.theme.global.colors['light-1']};
+      border: none;
+    `
+  }
+  border-radius: 4px;
+  margin-bottom: .8em;
+  margin-top: .8em;
+  box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.25);
+
+  &:hover {
+    ${props => props.theme.dark ?
+      css`
+        background: ${props.theme.global.colors['neutral-1']};
+        border: 2px solid ${props.theme.global.colors['light-1']};
+      ` :
+      css`
+        background: linear-gradient(180deg, ${props.theme.global.colors['light-2']} 0%, ${props.theme.global.colors['accent-1']} 100%);
+        border: none;
+      `
+    } 
+    box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.25);
+  }
+
+  &:active {
+    background: ${props => props.theme.global.colors['brand']};
+    ${props => props.theme.dark ? 
+      css`border: 2px solid ${props.theme.global.colors['light-1']};` :
+      css`border: 1px solid ${props.theme.global.colors['light-5']};`
+    }
+    box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.25);
+    color: white;
+  }
+
+  &:focus:not(:focus-visible) {
+    box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.25);
+  }
+`
+
+function StyledButtonLabel ({ color, count = 0, label }) {
+  const { t } = useTranslation('plugins')
+  const status = t('HighlighterTask.status', { count })
+
+  return (
+    <Box
+      align='center'
+      as='span'
+      direction='row'
+    >
+      <StyledLabelColor color={color} />
+      <Box
+        fill='horizontal'
+        direction='row'
+        justify='between'
+        margin={{ left: 'small' }}
+      >
+        <Text size='16px'>
+          {label}
+        </Text>
+        <Text color={count ? 'inherit' : 'light-4' }>
+          {status}
+        </Text>
+      </Box>
+    </Box>
+  )
+}
+
+export function HighlighterTask ({
+  annotation,
+  disabled = false,
+  task
+}) {
+  const { t } = useTranslation('plugins')
+
+  function createLabelAnnotation(selection, labelIndex) {
+    // currently we only deal with one selection at a time
+    const range = selection.getRangeAt(0)
+    const offset = getOffset(selection)
+    const start = offset + range.startOffset
+    const endOffset = extraNewLineCharacter(range)
+    const end = offset + endOffset
+    const labelInformation = getSnapshot(task.highlighterLabels[labelIndex])
+    const selectable = selectableArea(selection, range, offset, start, end)
+    
+    if (selectable) {
+      const newValue = Array.from(annotation.value)
+      newValue.push({
+        labelInformation,
+        start,
+        end,
+        text: range.toString()
+      })
+      annotation.update(newValue)
+    }
+
+    selection.collapseToEnd()
+  }
+
+  function handleClick (event, index) {
+    const selection = document.getSelection()
+    selection.rangeCount ? createLabelAnnotation(selection, index) : null
```

Comment:
I think this check is failing in Firefox. Clicking the buttons does nothing.

Reply:
Turns out Firefox wasn't reading `ariaLabel` on a parent node per the `selectableArea` helper as expected, [this commit](https://github.com/zooniverse/front-end-monorepo/pull/4370/commits/da5cb18f467b72fbd8d13a0b93e2f6da0d2ba29d#diff-40a9affb9a1951ca60e6c41e528c5e62aa577536938dc6485973890263ed1294L4) fixes.

Reply:
Noting the button to delete a selection needs better styling in Firefox and Safari.

Reply:
Oh yes, if you're working with a HTML element, rather than a React component, you should use `getAttribute` to get the ARIA label.
```js
  const ariaLabel = element.getAttribute('aria-label');
```

Reply:
Also, a GitHub markdown tip: commit hashes are automatically converted to links: eg. da5cb18f467b72fbd8d13a0b93e2f6da0d2ba29d